platform/solr/solr-factory/src/main/java/org/codice/solr/factory/SolrServerFactory.java

```diff
@@ -435,26 +430,44 @@ private static boolean solrCoreExists(SolrServer solrServer, String coreName) {
 
         private final String coreUrl;
 
-        private int retryCount;
+        private int attemptCount;
 
         public SolrServerFetcher(String url, String coreName, String configFile, String coreUrl) {
             this.url = url;
             this.coreName = coreName;
             this.configFile = configFile;
             this.coreUrl = coreUrl;
-            this.retryCount = 0;
+            this.attemptCount = 0;
         }
 
         @Override
         public SolrServer call() throws Exception {
-            while (true) {
+            while (attemptCount < MAX_ATTEMPTS) {
+                ++attemptCount;
+
                 try {
+                    LOGGER.debug("Attempting to get SolrServer with parameters:\n" + "\turl: {}\n"
+                                    + "\tcore name: {}\n" + "\tconfig file: {}\n" + "\tcore url: {}\n"
+                                    + "Attempt number {}",
+                            url,
+                            coreName,
+                            configFile,
+                            coreUrl,
+                            attemptCount);
                     return getSolrServer(url, coreName, configFile, coreUrl);
                 } catch (Exception e) {
-                    retryCount = Math.min(retryCount + 1, MAX_RETRY_COUNT);
-                    Thread.sleep((long) Math.pow(2, Math.min(retryCount, MAX_RETRY_COUNT)) * 50);
+                    LOGGER.warn("Error retrieving SolrServer.", e);
+
+                    long waitTime = (long) Math.pow(2, attemptCount) * RETRY_INTERVAL_MULTIPLIER;
+                    LOGGER.debug("Waiting {} milliseconds until the next attempt.", waitTime);
+                    Thread.sleep(waitTime);
                 }
             }
+
+            String exceptionText = String.format(
+                    "Exceeded maximum number of attempts (%d) to get a SolrServer.",
```

Comment:
I think I misunderstood the intent of the original code because it used a variable named `MAX_RETRY_COUNT`. The tests fail because they aren't passing a valid Solr URL to the factory,  which causes the tests to hang indefinitely because the number of retries is uncapped. Simply fixing the URL + adding a timeout to the tests should be sufficient.